src/access/RequestPath.h

```diff
@@ -19,15 +19,29 @@
 #pragma once
 
 #include <lib/core/DataModelTypes.h>
+#include <optional>
 
 namespace chip {
 namespace Access {
 
+enum class RequestType : uint8_t
+{
+    kRequestTypeUnknown,
+    kReadRequest,
+    kWriteRequest,
+    kInvokeRequest,
+    kSubscribeEventRequest
+};
+
 struct RequestPath
 {
     // NOTE: eventually this will likely also contain node, for proxying
-    ClusterId cluster   = 0;
-    EndpointId endpoint = 0;
+    ClusterId cluster       = 0;
+    EndpointId endpoint     = 0;
+    RequestType requestType = RequestType::kRequestTypeUnknown;
+
+    //entityId represents an attribute, command, or event ID, which is determined by the requestType. Wildcard if omitted.
```

Comment:
OK, so what request type is used for an event read (not subscribe)?

Reply:
The spec only considers restrictions on events for subscribe, not read.  If that ends up being a gap in the spec, we would address here after spec changes (which are now merged).

Reply:
Fundamentally, "read" vs "subscribe" is not a useful distinction for access purposes.  The distinctions we care about are:

1. What does the entity id represent?  Is it an attribute, event, or command?
2. Which access type are we performing?  Really, for events and commands there is only one access type (read and invoke respectively), while for attributes there are potentially two: read and write.

Reply:
This should be resolved.